### PR TITLE
#206: Split current code into common and spark-2.4 modules

### PR DESCRIPTION
**Pull request by utnaf**
In order to be able to add Spark 3.0 support (#206) we are first opening a PR where we introduce maven modules project structure. Once this is merged we'll add the Spark 3.0 module. 

Current problems:

- [x] <del>When packaged, a `doc` directory is created inside each module (maven frontend plugin's fault)</del>

### CODE REVIEW

**Comment by moxious** [comment]
needs review -- does this keep antora build pipeline in good shape?

**Comment by moxious** [comment]
@conker84 to review and merge

**Review by conker84 — APPROVED** [review]
@utnaf generally looks good, IDK if makes sense implement my suggestion

**Comment by conker84** [comment]
ok for some reason seems that GitHub ignored my comment, I'll post it here:

IDK if makes sense to create a `test-support` module like we did in Streams:
https://github.com/neo4j-contrib/neo4j-streams/tree/4.0/test-support
In order to reuse test classes like `TestUtils` and `Neo4jContainerExtension` on all modules

**Comment by conker84** [comment]
Thinking about it, it makes totally sense there are duplications like `Neo4jContainerExtension` that we want to avoit, please add the `test-support module`

**Review by conker84 — CHANGES_REQUESTED** [review]
se the comment above

**Comment by utnaf** [comment]
@conker84 updated

File: spark-2.4/pom.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        <dependency>
+            <groupId>neo4j-contrib</groupId>
+            <artifactId>neo4j-connector-apache-spark_${scala.binary.version}_common</artifactId>
+            <version>4.0.0</version>
+        </dependency>
+
+        <dependency>
+            <groupId>neo4j-contrib</groupId>
+            <artifactId>neo4j-connector-apache-spark_${scala.binary.version}_test-support</artifactId>
+            <version>4.0.0</version>
```
Comment (conker84) [inline spark-2.4/pom.xml:34]:
```suggestion
            <version>4.0.0</version>
            <scope>test</test>
```

**Review by conker84 — CHANGES_REQUESTED** [review]
Just that change we need to import it as test package otherwise we'll include them into the distribution artifact.

**Comment by conker84** [comment]
LGTM

I'm merging this so we can move forward to `3.0` integration, thank you @utnaf

**Review by conker84 — APPROVED** [review]

**Pull request merged**